### PR TITLE
Enable `.indent_mods` argument in all analysis functions

### LINKED ISSUE #941: Standardize `.indent_mods` argument across all analysis functions
Check all analysis functions for the `.indent_mod` argument and add/enable where needed.

### PR DESCRIPTION
**Pull request by edelarua**
Closes #941 

Indentation implemented in all functions except `tabulate_survival_subgroups` and `tabulate_rsp_subgroups`.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -2,9 +2,11 @@
 
 ### Enhancements
 * Added explicit zero counts to `g_km` plot "at risk" annotation tables.
+* Implemented `.indent_mods` argument in functions `h_tab_one_biomarker`, `h_tab_rsp_one_biomarker`, `h_tab_surv_one_biomarker`, `summarize_logistic`, `logistic_summary_by_flag`, `tabulate_rsp_biomarkers`, a_coxreg, `summarize_coxreg`, `tabulate_survival_biomarkers`, `surv_time`, `surv_timepoint`, and `cfun_by_flag`.
 
 ### Miscellaneous
 * Updated README to include installation instructions for CRAN.
+* Began deprecation of `indent_mod` argument which is replaced by the `.indent_mods` argument in `summarize_num_patients` and `analyze_num_patients`.
```
Comment (Melkiades) [inline NEWS.md:9]:
```suggestion
* Began deprecation of `indent_mod` argument and replaced by the `.indent_mods` argument in `summarize_num_patients` and `analyze_num_patients`.
```

File: R/argument_convention.R [hunk 2]
```diff
@@ -27,10 +27,7 @@
 #'   Note that in that case the remaining occurrence levels in the table are sorted alphabetically.
 #' @param id (`string`)\cr subject variable name.
 #' @param is_event (`logical`)\cr `TRUE` if event, `FALSE` if time to event is censored.
-#' @param indent_mod (`count`)\cr it can be negative. Modifier for the default indent position for the
-#'   structure created by this function(subtable, content table, or row) and
-#'   all of that structure's children. Defaults to 0, which corresponds to the
-#'   unmodified default behavior.
```
Comment (Melkiades) [inline R/argument_convention.R:33]: I think this information should be preserved, maybe in a shorter form

File: R/compare_variables.R [hunk 3]
```diff
@@ -360,6 +360,9 @@ a_compare.logical <- make_afun(
 #' Constructor function which creates a combined formatted analysis function.
 #'
 #' @inheritParams argument_convention
+#' @param .indent_mods (named `vector` of `integer`)\cr indent modifiers for the labels. Each element of the vector
```
Comment (Melkiades) [inline R/compare_variables.R:363]: why overloading the indent_mods definition from argument_convention?

Comment (edelarua) [reply R/compare_variables.R:363]: `.indent_mods` has different behaviour/requires different input for functions `compare_vars`, `summarize_vars`, `summarize_colvars`, `surv_time`, and `surv_timepoint`.

**Pull request merged**